File: src/modules/procurement/controllers/procurement.controller.ts
import {
  Controller,
  Get,
  Param,
  ParseUUIDPipe,
  UseGuards,
} from '@nestjs/common';
import { ApiBearerAuth, ApiOperation, ApiTags } from '@nestjs/swagger';
import { MinRole } from '../../auth/decorators/min-role.decorator';
import { JwtAuthGuard } from '../../auth/guards/jwt-auth.guard';
import { RolesGuard } from '../../auth/guards/roles.guard';
import { EmployeeRole } from '../../employee/enums/employee-role.enum';
import { RequestService } from '../../request/services/request.service';
import { ProcurementService } from '../services/procurement.service';

@ApiTags('procurement')
@ApiBearerAuth()
@UseGuards(JwtAuthGuard, RolesGuard)
@MinRole(EmployeeRole.PROCUREMENT_MANAGER)
@Controller('procurement')
export class ProcurementController {
  constructor(
    private readonly procurementService: ProcurementService,
    private readonly requestService: RequestService,
  ) {}

  @Get('approvals')
  @ApiOperation({
    summary: 'List pending procurement approvals',
    description: 'Returns requests awaiting procurement manager review.',
  })
  pendingApprovals() {
    return this.requestService.findProcurementPending();
  }

  @Get('requests/:id/availability')
  @ApiOperation({
    summary: 'Check request asset availability',
    description:
      'Checks whether available inventory exists for a loan request equipment model.',
  })
  checkAvailability(@Param('id', ParseUUIDPipe) id: string) {
    return this.procurementService.checkAvailability(id);
  }
}

File: src/EquipmentApi/Controllers/ProcurementController.cs
using EquipmentApi.Authorization;
using EquipmentApi.Enums;
using EquipmentApi.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace EquipmentApi.Controllers;

[ApiController]
[Authorize]
[MinRole(EmployeeRole.ProcurementManager)]
[Route("procurement")]
public class ProcurementController : ControllerBase
{
    private readonly ProcurementService _procurementService;
    private readonly RequestService _requestService;

    public ProcurementController(ProcurementService procurementService, RequestService requestService)
    {
        _procurementService = procurementService;
        _requestService = requestService;
    }

    [HttpGet("approvals")]
    public Task<List<Entities.EquipmentRequest>> PendingApprovals(CancellationToken cancellationToken) =>
        _requestService.FindProcurementPendingAsync(cancellationToken);

    [HttpGet("requests/{id:guid}/availability")]
    public Task<object> CheckAvailability(Guid id, CancellationToken cancellationToken) =>
        _procurementService.CheckAvailabilityAsync(id, cancellationToken);
}
